linkerd/app/outbound/src/http/logical/policy/route/metrics/tests.rs

```diff
@@ -99,6 +99,127 @@ async fn http_request_statuses() {
     assert_eq!(mixed.get(), 1);
 }
 
+#[tokio::test(flavor = "current_thread", start_paused = true)]
+async fn http_request_hostnames() {
+    const HOST_1: &str = "great.website";
+    const URI_1_1: &str = "https://great.website/path/to/index.html#fragment";
+    const URI_1_2: &str = "https://great.website/another/index.html";
+    const HOST_2: &str = "different.website";
+    const URI_2: &str = "https://different.website/index.html";
+    const URI_3: &str = "https://[3fff::]/index.html";
```

Comment:
:point_up: tests! we show that our labels look right when sending traffic to different hosts, and to places with an opaque address.